### PR TITLE
[AMORO-2226]: Rename Flink Model Arctic to Amoro

### LINKED ISSUE #2226: [Improvement]: Rename Flink Model Arctic to Amoro
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

ArcticCatalogFactoryOptions IDENTIFIER Rename to Amoro And Java FileName Rename to Amoro

NOTE: This is an upgrade that may potentially break forward compatibility

### How should we improve?

- Rename Flink ArcticCatalog to AmoroCatalog
- Rename Flink Model Code Arctic to Amoro
- Rename Flink Model Test Code

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by czy006**
## Why are the changes needed?

Close #2226 .

## Brief change log

- rename flink mode java file name prefix to amoro

## How was this patch tested?

- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible
- [ ] Add screenshots for manual tests if appropriate
- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? ( no)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Comment by xleoken** [comment]
👍

File: amoro-mixed-format/amoro-mixed-format-flink/amoro-mixed-format-flink-common/src/main/java/org/apache/amoro/flink/read/source/AmoroScanContext.java [hunk 1]
```diff
@@ -38,14 +38,14 @@
 import java.util.Objects;
 
 /** This is an arctic source scan context. */
```
Comment (xieyi888) [inline amoro-mixed-format/amoro-mixed-format-flink/amoro-mixed-format-flink-common/src/main/java/org/apache/amoro/flink/read/source/AmoroScanContext.java:40]:
```suggestion
/** This is an amoro source scan context. */
```

File: amoro-mixed-format/amoro-mixed-format-flink/amoro-mixed-format-flink-common/src/main/java/org/apache/amoro/flink/read/AmoroSource.java [hunk 2]
```diff
@@ -42,29 +42,29 @@
...
   private static final long serialVersionUID = 1L;
-  private final ArcticScanContext scanContext;
+  private final AmoroScanContext scanContext;
   private final ReaderFunction<T> readerFunction;
   private final TypeInformation<T> typeInformation;
-  private final ArcticTableLoader loader;
+  private final AmoroTableLoader loader;
   private final String tableName;
   /**
    * generate arctic watermark. This is only for lookup join arctic table, and arctic table is used
```
Comment (xieyi888) [inline amoro-mixed-format/amoro-mixed-format-flink/amoro-mixed-format-flink-common/src/main/java/org/apache/amoro/flink/read/AmoroSource.java:60]:
```suggestion
   * generate amoro watermark. This is only for lookup join amoro table, and amoro table is used
```

**Comment by xieyi888** [comment]
It seems variable, method, comments  which contains arctic also need to be renamed to amoro in  new PR in the future?

**Comment by czy006** [comment]
Yes this pr just renames some classes, if we need to change some of the configuration from arctic to amoro, do we have to break some compatibility @xieyi888

**Comment by zhoujinsong** [comment]
Thanks for the contribution! 

But I don't think renaming arctic to amoro in all places is a good solution here.
The new project name amoro means a lot more than just the mixed-format now.
So I tried to resolve this issue by #2848 , can you help to review it? @czy006 @xieyi888

**Comment by zhoujinsong** [comment]
Completed by #2848

**Pull request closed without merging**